packages/grid/x-data-grid/src/hooks/core/useGridApiInitialization.ts

```diff
@@ -59,30 +103,15 @@ export function useGridApiInitialization<
   props: Pick<DataGridProcessedProps, 'signature'>,
 ): React.MutableRefObject<PrivateApi> {
   const publicApiRef = React.useRef() as React.MutableRefObject<Api>;
-
-  if (!publicApiRef.current) {
-    const state = {} as Api['state'];
-
-    publicApiRef.current = {
-      state,
-      store: Store.create(state),
-      instanceId: { id: globalId },
-    } as Api;
-
-    globalId += 1;
-  }
-
   const privateApiRef = React.useRef() as React.MutableRefObject<PrivateApi>;
-  if (!privateApiRef.current) {
-    privateApiRef.current = wrapPublicApi<PrivateApi, Api>(publicApiRef.current);
 
-    privateApiRef.current.register('private', {
-      caches: {} as PrivateApi['caches'],
-      eventManager: new EventManager(),
-    });
+  if (!privateApiRef.current) {
+    privateApiRef.current = createPrivateAPI(publicApiRef) as PrivateApi;
   }
 
-  React.useImperativeHandle(inputApiRef, () => publicApiRef.current, [publicApiRef]);
+  if (!publicApiRef.current || !(publicApiRef.current as any)[SYMBOL_API_TYPE]) {
```

Comment:
What is the purpose of this condition?
```ts
!(publicApiRef.current as any)[SYMBOL_API_TYPE]
```

Reply:
Leftover, it can go now.

Reply:
It looks like we can get rid of `SYMBOL_API_TYPE` since it's not being used anymore

Reply:
Done